src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -347,6 +356,58 @@ private static void populateLambdaArgs(Class<?> functionalInterface, final Class
     }
   }
 
+  private static final class MethodRefOffsetResolutionFailed extends RuntimeException {}
+
+  /**
+   * Resolves method ref offset.
+   */
+  private static int resolveMethodRefOffset(ConstantPool constantPool, Class<?> lambdaType) {
+    Integer offset = methodRefOffsetCache.get();
+
+    if (offset == null) {
+      int constantPoolSize = constantPool.getSize();
+
+      for (int i = 0; i < constantPoolSize; i++) {
+        try {
+          constantPool.getMemberRefInfoAt(constantPoolSize - i);
+          offset = i;
+          break;
+        } catch (IllegalArgumentException ignore) {
+        }
+      }
+
+      if (offset == null)
+        offset = -1;
+    }
+
+    if (CACHE_ENABLED)
+      methodRefOffsetCache.compareAndSet(null, offset);
+
+    if (offset >= 0)
+      return offset;
+    else
+      throw new MethodRefOffsetResolutionFailed();
+  }
+
+  /**
+   * Resolves method ref offset of method reference lambda type having autoboxed return type.
+   * We can't cache the result value because results are different depending on given {@code lambdaType}.
+   */
+  private static int resolveAutoboxedMethodRefOffset(ConstantPool constantPool, Class<?> lambdaType) {
+    int constantPoolSize = constantPool.getSize();
+
+    for (int i = resolveMethodRefOffset(constantPool, lambdaType) + 1; i < constantPoolSize; i++) {
+      try {
+        // ignore constructor
+        if (!constantPool.getMemberRefInfoAt(constantPoolSize - i)[1].equals("<init>"))
```

Comment:
What is the intuition for wanting to ignore the constructor in the autoboxed method ref case but not in the other case? Shouldn't we always ignore the `<init>` method's member ref info? Forgive me if this is a naive question.